tiatoolbox/utils/misc_utils.py

```diff
@@ -0,0 +1,48 @@
+"""
+This file contains miscellaneous small functions repeatedly required and used in the repo
+"""
+import os
+import pathlib
+
+
+def split_path_name_ext(full_path):
+    """
+    Split path of a file to directory path, file name and extension
+
+    Args:
+        full_path: Path to a file
+
+    Returns:
+        input_dir: directory path
+        file_name: name of the file without extension
+        ext: file extension
```

Comment:
The Returns statement here isn't in a valid Google docstring format. As far as I can tell it should be either just a description or a type: description pair (optionally followed by a longer description):

Examples from https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html:

```
Returns: True if successful, False otherwise.
```

```
Returns:
    True if successful, False otherwise.
```

```
Returns:
    bool: True if successful, False otherwise.
```

```
Returns:
    bool: True if successful, False otherwise.

    The return type is optional and may be specified at the beginning of
    the ``Returns`` section followed by a colon.

    The ``Returns`` section may span multiple lines and paragraphs.
    Following lines should be indented to match the first line.

    The ``Returns`` section supports any reStructuredText formatting,
    including literal blocks::
        {
            'param1': param1,
            'param2': param2
        }
```

Here a name is given instead of a type, in the documentation this will render as:

`split_path_name_ext(full_path):`
Split path of a file to directory path, file name and extension

**Return type** input_dir

It appears that only one key-value pair is valid (since python technically can only return one thing), so only the first pair will show here. To document as tuple as the return type I think it should read something like this:

```
Returns:
    Tuple[str, str, str]: Three sections of the input file path
    (input directroy path, file name, file extension)
```

based on https://bitbucket.org/birkenfeld/sphinx-contrib/issues/111/napoleon-multiple-returns-indentations which renders as

`split_path_name_ext(full_path):`
Split path of a file to directory path, file name and extension

**Return type** Tuple[str, str, str]

However, anything you put there will show. It looks like most (including [numpy](https://github.com/numpy/numpy/blob/master/numpy/core/numeric.py#L391-L462)) people do something like this:

```
Returns:
    tuple: Three sections of the input file path
    (input directory path, file name, file extension)
```

which renders as

**Returns** Three sections of the input file path (input directory path, file name, file extension)
**Return type** tuple

Reply:
Thanks for identifying this issue. Corrected in this commit https://github.com/TIA-Lab/tiatoolbox/commit/b9b6dd5f9940591a52441381911555f9348ae566.